kubernetes/istio.go

```diff
@@ -384,5 +384,8 @@ func GatewayAPIClasses() []config.GatewayAPIClass {
 	if len(result) == 0 {
 		result = append(result, config.GatewayAPIClass{Name: "Istio", ClassName: "istio"})
 	}
+	if ambientEnabled {
```

Comment:
We need to put this condition inside the above `if len(result) == 0 {` as when configuration is already set in 'ExternalServices.Istio.GatewayAPIClasses' then we cannot add any default value. Configured values are taken, default values are set when no config provided.